src/psycopmlutils/timeseriesflattener/data_integrity.py

```diff
@@ -0,0 +1,284 @@
+        CategoryMismatchTrainTest(
+            **kwargs
+        ).add_condition_new_category_ratio_less_or_equal(),
+        IndexTrainTestLeakage(**kwargs).add_condition_ratio_less_or_equal(),
+    )
+
+
+def label_split_checks() -> Suite:
+    """Deepchecks train/test validation suite for checks that require a label.
+
+    Returns:
+        Suite: a deepchecks Suite
+    """
+    return Suite(
+        "Split validation checks requiring labels",
+        FeatureLabelCorrelationChange()
+        .add_condition_feature_pps_difference_less_than()
+        .add_condition_feature_pps_in_train_less_than(),
+        TrainTestLabelDrift().add_condition_drift_score_less_than(),
+    )
+
+
+def load_split_predictors_and_outcomes(
+    path: Path,
+    split: str,
+    include_id: bool,
+    nrows: Optional[int] = None,
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Loads a given data split from a directory and returns predictors and
+    outcomes separately.
+
+    Args:
+        path (Path): Path to directory containing data files
+        split (str): Which split to load
+        include_id (bool): Whether to include 'dw_ek_borger' in predictor df
+        nrows (Optional[int]): Number of rows to load from each file.
+
+    Returns:
+        Tuple[pd.DataFrame, pd.DataFrame]: Tuple where first element is the
+        predictors and second element the outcomes
+    """
+    split = load_split(path, split, nrows=nrows)
+    predictors, outcomes = separate_predictors_and_outcome(split, include_id=include_id)
+    return predictors, outcomes
+
+
+def separate_predictors_and_outcome(
+    df: pd.DataFrame,
+    include_id: bool,
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Split predictors and outcomes into two dataframes. Assumes predictors to
+    be prefixed with 'pred', and outcomes to be prefixed with 'outc'. Timestamp
+    is also returned for predictors, and optionally also dw_ek_borger.
+
+    Args:
+        df (pd.DataFrame): Dataframe containing generates features
+        include_id (bool): Whether to include 'dw_ek_borger' in predictor df
+
+    Returns:
+        Tuple[pd.DataFrame, pd.DataFrame]: Tuple where first element is the
+        predictors and second element the outcomes
+    """
+    pred_regex = (
+        "^pred|^timestamp" if not include_id else "^pred|^timestamp|dw_ek_borger"
+    )
+    predictors = df.filter(regex=pred_regex)
+    outcomes = df.filter(regex="^outc")
+    return predictors, outcomes
+
+
+def load_split(path: Path, split: str, nrows: Optional[int] = None) -> pd.DataFrame:
+    """Loads a given data split as a dataframe.
+
+    Args:
+        path (Path): Path to directory containing data files
+        split (str): Which string to look for (e.g. 'train', 'val', 'test')
+        nrows (Optional[int]): Whether to only load a subset of the data
+
+    Returns:
+        pd.DataFrame: The loaded dataframe
+    """
+    return pd.read_csv(list(path.glob(f"*{split}*"))[0], nrows=nrows)
```

Comment:
Seems like some of this code is more about file manipulation, and thus reuseable! E.g. "load_split" seems awesome. Especially if we can generalise it with a "most_recent" flag or similar.

Perhaps extract them to a separate "utils" file?

Reply:
moved to `psycopmlutils.loaders.flattened`

Reply:
Let's JIT `most_recent` ;)